docs/src/content/docs/tutorials/url-shortener.md

```diff
@@ -0,0 +1,176 @@
+    }
+
+    var db *dicedb.Client
+
+    // Initialize DiceDB connection
+    func init() {
+        db = dicedb.NewClient(&dicedb.Options{
+            Addr: "localhost:7379",
+        })
+    }
+
+    // Creates a short URL from a given long URL
+    func CreateShortURL(c *gin.Context) {
+        var requestBody URL
+        if err := c.ShouldBindJSON(&requestBody); err != nil {
+            c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid request"})
+            return
+        }
+
+        // Generate unique short ID and construct the short URL
+        shortID := uuid.New().String()[:8]
+        requestBody.ID = shortID
+        requestBody.ShortURL = "http://localhost:8080/" + shortID
+
+        // Serialize URL struct to JSON and store it in DiceDB
+        urlData, err := json.Marshal(requestBody)
+        if err != nil {
+            c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to save URL"})
+            return
+        }
+
+        if err := db.Set(context.Background(), shortID, urlData, 0).Err(); err != nil {
+            c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to save URL"})
+            return
+        }
+
+        c.JSON(http.StatusCreated, gin.H{"short_url": requestBody.ShortURL})
+    }
+
+    // Redirects to the original URL based on the short URL ID
+    func RedirectURL(c *gin.Context) {
+        id := c.Param("id")
+
+        // Retrieve stored URL data from DiceDB
+        urlData, err := db.Get(context.Background(), id).Result()
+        if err != nil {
+            c.JSON(http.StatusNotFound, gin.H{"error": "URL not found"})
+            return
+        }
+
+        // Deserialize JSON data back into URL struct
+        var url URL
+        if err := json.Unmarshal([]byte(urlData), &url); err != nil {
+            c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to decode URL data"})
+            return
+        }
+
+        // Redirect user to the original long URL
+        c.Redirect(http.StatusFound, url.LongURL)
+    }
+
+    func main() {
+        router := gin.Default()
+
+        // Define endpoints for creating short URLs and redirecting
+        router.POST("/shorten", CreateShortURL)
+        router.GET("/:id", RedirectURL)
+
+        // Start the server on port 8080
+        if err := router.Run(":8080"); err != nil {
+            log.Fatal("Failed to start server:", err)
+        }
+    }
+    ```
+
+## Explanation of Key Parts
+1. Database Initialization: The `init()` function sets up a DiceDB client to connect to `localhost:7379`.
+2. `CreateShortURL` Endpoint: Handles the `/shorten` route. It generates a unique ID, constructs the short URL, serializes the URL data, and saves it in DiceDB.
+3. `RedirectURL` Endpoint: Handles the `/:id` route. It retrieves the original URL by the short ID from DiceDB and redirects the user to it.
+4. Starting the Server: The `main` function starts the Gin server on port `8080`.
```

Comment:
We should make use of subheadings to make this clearer. PTAL at the following approach to write this:
```suggestion

### 1. Initialize the DiceDB Client

We set up the DiceDB client in the `init` function:

    ```go
db = dicedb.NewClient(&dicedb.Options{
    Addr: "localhost:7379",
})
    ```

### 2. Create Short URL Endpoint

- **Input Validation**: Ensures the `long_url` field is present.
- **Short ID Generation**: Uses `uuid` to create a unique 8-character ID.
- **Data Serialization**: Converts the `URL` struct to JSON.
- **Data Storage**: Saves the JSON data in DiceDB with the `Set` command.
- **Response**: Returns the generated short URL.

### 3. Redirect to Original URL Endpoint

- **Data Retrieval**: Fetches the URL data from DiceDB using the `Get` command.
- **Data Deserialization**: Converts JSON back to the `URL` struct.
- **Redirection**: Redirects the user to the `LongURL`.

### 4. Start the Server

The `main` function sets up the routes and starts the server on port `8080`.
```